src/objects/zcl_abapgit_object_iaxu.clas.abap

```diff
@@ -0,0 +1,216 @@
+
+  PUBLIC SECTION.
+    INTERFACES zif_abapgit_object.
+
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+    DATA: mo_xml_api           TYPE REF TO cl_w3_api_xml3,
+          mc_source_style_2006 TYPE w3style VALUE 'XML',
+          mc_generator_class   TYPE w3styleclass VALUE 'CL_ITS_GENERATE_XML3'.
+
+    METHODS:
+      read
+        EXPORTING es_attr TYPE w3tempattr
+        RAISING   zcx_abapgit_exception,
+      save
+        IMPORTING is_attr TYPE w3tempattr
+        RAISING   zcx_abapgit_exception.
+
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_object_iaxu IMPLEMENTATION.
+
+
+  METHOD read.
+
+    DATA: li_service TYPE REF TO cl_w3_api_xml3,
+          lv_name    TYPE iacikeyt.
+
+
+    lv_name = ms_item-obj_name.
+
+    cl_w3_api_xml3=>load(
+      EXPORTING
+        p_xml_name          = lv_name
+      IMPORTING
+        p_xml               = mo_xml_api
+        p_attributes        = es_attr
+      EXCEPTIONS
+        object_not_existing = 1
+        permission_failure  = 2
+        data_corrupt        = 3
+        error_occured       = 4
+        OTHERS              = 5 ).
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( 'error from w3_api_xml3~load' ).
+    ENDIF.
+
+    CLEAR: es_attr-chname,
+           es_attr-tdate,
+           es_attr-ttime,
+           es_attr-devclass.
+
+  ENDMETHOD.
+
+
+  METHOD save.
+
+    cl_w3_api_xml3=>create_new(
+      EXPORTING
+        p_source_style_2006       = mc_source_style_2006
+        p_xml_data                = is_attr
+        p_generator_class         = mc_generator_class
+        p_program_name            = is_attr-programm
+      IMPORTING
+        p_xml                     = mo_xml_api
+      EXCEPTIONS
+        undefined_name            = 1
+        error_occured             = 2
+        object_already_existing   = 3
+        not_authorized            = 4
+        action_cancelled          = 5
+        OTHERS                    = 6 ).
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error from w3_api_xml3~create_new| ).
+    ENDIF.
+
+    mo_xml_api->if_w3_api_object~save( ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+    rv_user = c_user_unknown. " todo
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lo_xml_api TYPE REF TO cl_w3_api_xml3,
+          lv_name    TYPE iacikeyt.
+
+
+    lv_name = ms_item-obj_name.
+
+    cl_w3_api_xml3=>load(
+      EXPORTING
+        p_xml_name          = lv_name
+      IMPORTING
+        p_xml               = lo_xml_api
+      EXCEPTIONS
+        object_not_existing = 1
+        permission_failure  = 2
+        data_corrupt        = 3
+        error_occured       = 4
+        OTHERS              = 5 ).
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( 'error from w3_api_xml3~load' ).
+    ENDIF.
+
+    lo_xml_api->if_w3_api_object~set_changeable( abap_true ).
+    lo_xml_api->if_w3_api_object~delete( ).
+    lo_xml_api->if_w3_api_object~save( ).
```

Comment:
These methods raise a bunch exceptions. Suggest to handle them.

Reply:
I agree. Many other classes (e.g. IATU or IASP to just name some) are to be reworked then too, right?

Reply:
Opened issue #2982